src/script_lua.c

```diff
@@ -1430,52 +1430,65 @@ void luaRegisterLogFunction(lua_State* lua) {
     lua_settable(lua,-3);
 }
 
-void luaRegisterRedisAPI(lua_State* lua) {
+/*
+ * Adds server.* functions/fields to lua such as server.call etc. 
+ * This function only handles fields common between Functions and LUA scripting. 
+ * scriptingInit() and functionsInit() may add additional fields specific to each.   
+ */
+void luaRegisterServerAPI(lua_State* lua) {
+    /* In addition to registering server.call/pcall API, we will throw a customer message when a script accesses undefined global variable. 
+     * LUA stored global variables in the global table, accessible to us on stack at virtual index = LUA_GLOBALSINDEX. 
+     * We will set __index handler in global table's metatable to a custom C function to achieve this - handled by luaSetAllowListProtection.  
+     * Refer to https://www.lua.org/pil/13.4.1.html for documentation on __index and https://www.lua.org/pil/contents.html#13 for documentation on metatables.  
+     * We need to pass global table to lua invocations as parameters. To achieve this, lua_pushvalue invocation brings global variable table to the top of 
+     * the stack by pushing value from global index onto the stack. And lua_pop invocation after luaSetAllowListProtection removes it - resetting the stack to its original state. */ 
     lua_pushvalue(lua, LUA_GLOBALSINDEX);
     luaSetAllowListProtection(lua);
     lua_pop(lua, 1);
 
+    /* Add default C functions provided in deps/lua codebase to handle basic data types such as table, string etc. */
     luaLoadLibraries(lua);
 
+    /* Before Valkey 7, LUA used to return error messages as strings from pcall function. With Valkey 7, LUA now returns error messages as tables. 
```

Comment:
In other places, we have decided to not mention pre-7 history. The origin of Valkey is 7.2.5. Another option could be retaining "Redis", i.e., "Before Redis 7"

```suggestion
    /* LUA returns error messages as tables. 
```

Reply:
But then how do I document pre-7 behavior? We still have code to handle that behavior. I prefer to keep any compatibility code documented. But I can change the language.

Reply:
We refer to old versions as "Redis OSS" (according to redis trademark policy).

"Before Redis OSS 7"

Reply:
@zuiderkwast I incorporated your change in new PR.